packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -357,6 +359,10 @@ class DateTimePicker extends React.Component {
     const isDateValid = DateUtil.isValidDate(formattedDate, 'YYYY-MM-DD');
     const isTimeValid = DateTimeUtils.isValidTime(this.timeValue, this.props.showSeconds);
 
+    if (isDateValid) {
+      this.dateValue = DateUtil.formatISODate(date, this.state.dateFormat);
```

Comment:
Is line 354 (`this.dateValue = event.target.value;`) still needed since this.dateValue needs to be constructed using the values from all three inputs.

Reply:
I removed the check at line 354 and did some local testing and as of now everything seems to be working as expected.
I have removed the check here : https://github.com/cerner/terra-framework/pull/1208/commits/b3e858cfda4d1b7e6f4466ab9f32e863bd2132c0